sdk/core/src/main/java/com/klaviyo/coresdk/networking/requests/TrackRequest.kt

```diff
@@ -1,64 +1,61 @@
  *
- * @property customerProperties map of customer information we will be using to identify the user
- * @property properties map of property information we will be attaching to this request
+ * @constructor apiKey - the API key to identify this request
+ * @constructor event - the [KlaviyoEvent] to track
+ * @constructor customerProperties - map of customer information we will be using to identify the user
+ * @constructor properties - map of property information we will be attaching to this request
  *
  * @property timestamp The time that this event occurred
  * @property urlString The URL needed to reach the track API in Klaviyo
  * @property requestMethod [RequestMethod] determines the type of request that track requests are made over
  */
-internal class TrackRequest (
-        private var event: String,
-        private var customerProperties: KlaviyoCustomerProperties,
-        private var properties: KlaviyoEventProperties? = null
-): KlaviyoRequest() {
+internal class TrackRequest(
+    apiKey: String,
+    event: KlaviyoEvent,
+    customerProperties: KlaviyoCustomerProperties,
+    properties: KlaviyoEventProperties? = null
+) : KlaviyoRequest() {
     internal companion object {
-        const val TRACK_ENDPOINT = "api/track"
+        const val TRACK_ENDPOINT = "client/events"
     }
 
-    private var timestamp: Long? = null
-
     override var urlString = "$BASE_URL/$TRACK_ENDPOINT"
-    override var requestMethod = RequestMethod.GET
+    override var requestMethod = RequestMethod.POST
 
-    /**
-     * Builds a JSON payload suitable for a track request and returns it as a String
-     * Appends external information to the customer properties map before serializing it to JSON
-     *
-     * For more information on the structure of Klaviyo requests please reference the API docs:
-     * https://www.klaviyo.com/docs
-     *
-     * @return JSON payload as a string
-     */
-    override fun buildKlaviyoJsonQuery(): String {
-        customerProperties.setAnonymousId()
-        
-        val json = JSONObject(
-            mapOf(
-                "token" to KlaviyoConfig.apiKey,
-                "event" to event,
-                "customer_properties" to JSONObject(customerProperties.toMap()),
-                "properties" to properties?.let { JSONObject(properties?.toMap()) },
-                "time" to timestamp?.let { it }
-            ).filterValues { it != null }
-        )
+    override var queryData: Map<String, String> = mapOf(
+        "company_id" to apiKey
+    )
 
-        return json.toString()
-    }
+    override var payload: String? =
+        JSONObject(
+            mapOf(
+                "data" to mapOf(
+                    "type" to "event",
+                    "attributes" to mapOf(
+                        "metric" to mapOf(
+                            "name" to event.name,
+                        ),
+                        "profile" to JSONObject(customerProperties.setAnonymousId().toMap()),
```

Comment:
Do you need these sub json objects? Maybe can just do it at the top level?

Reply:
maybee...can revisit this later